Created the class Note for a single note.

syaya123 opened the pull request:
Created the metods: __str__, add_tags and change_note.
In the module class_note the bot initializes an object of the Notes class and creates list of tags; adds a new tag or tags to list of tags and also takes into account the limit of the number of tags; rewrites note or changes some words in the note.

syaya123 committed:
Created the class Note for a single note.

Created the metods: __str__, add_tags and change_note.

DenysTantsiura commented on .vscode/launch.json:
```diff
@@ -0,0 +1,18 @@
+{
```
Це зайвий файл, до нашої програми не відноситься, він для роботи ВСКода, в ГІТХАБ-десктопі його можна виключити для відстежування і він не потраплятиме більше до ГІТа.

syaya123 replied:
Видалю пізніше.

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+
+    def add_tags(self, new_tags: str) -> tuple:
+        '''add_tags...": The bot adds a new tag or tags to list of tags. 
+        The bot takes into account the limit of the number of tags.
```
Ліміт? Який? і звідки бере? Чи треба додати і додаватиметься? Ага нижче бачу константне значення 5

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+        Returns:
+            string(str): If false - the answer for the user, otherwise - a list of tags.'''
+
+        add_tags = new_tags.split(', ')
```
ну, якщо так, то так, треба буде десь повідомляти як, можна вводити і через пробіл, і через кому безпробілу...

syaya123 replied:
Треба буде вибрати, можна й через #

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+            string(str): If false - the answer for the user, otherwise - a list of tags.'''
+
+        add_tags = new_tags.split(', ')
+        number_tags = 5
```
Якщо не будемо вводити, і не змінюватиметься, то константи всі в апперкейсах, всі великі літери, начебто ж так було?

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+            if len(add_tags) <= delta:
+                for tag in add_tags:
+                    if tag in self.tags:
+                        return ('This tag already exists!')
```
Отуто ми в циклі, і при першому тезі в обох випадках(чи вже є чи ще немає такого тегу) ретюрн - зупинить цикл, тобто лише перший з декількох оброблюватиметься. Можна просто принтити

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+                    if tag in self.tags:
+                        return ('This tag already exists!')
+                    else:   
+                        self.tags.append(', '.join(add_tags))
```
self.tags.append(', '.join(add_tags))   тобто до спискутегів примірника self.tags додаємо СТРОКУ! в якій о'бєднуються елементи введеного списку, на кожній ітерації циклу (якщо без зупинки ретютном).
Тобто якщо всі нові теги - додаємо їх і повертаємо - все Ок, якщо хоч один тег вже занесений раніше - зупиняємо цикл і не перевіряємо наступні, бо ретюрним 'This tag already exists!', так і задумувалось?
Чи може додавати по одному тегу до списку, як окремі елементи?

DenysTantsiura commented on class_note.py:
```diff
@@ -0,0 +1,54 @@
+            else:
+                return (f'You can add only {delta} tags!')
+   
+    def change_note(self, changed_text, new_text) -> tuple:
```
значить повертає не тапл, виправити, і
примушувати користувача вводити весь текст великої нотатки, якщо він бажає її змінити... може трошки інакше зробити, прінтити тест старої нотатки і хай вводить новий текст для неї, якщо якісь зміни є, бо це не зміна а заміна в нотатках виходить. Можна обговорити на міті.

DenysTantsiura reviewed (changes requested):
Чудово, трошки підправити і обговорити...

DenysTantsiura commented:
Так, УВАЖНІШЕ! пише що хоче залиавати запит на main гілку, а треба в dev

syaya123 committed:
Update class_note

Rewrote uppercase constant and def add_tags (updated loop for: changed return ('This tag already exists!') on print and rewrote self.tags.append without join.
Changed the tuple to str in def change_note and removed the part about changing some words.

DenysTantsiura committed:
Delete launch.json from GitHub

DenysTantsiura committed:
Delete .gitignore
